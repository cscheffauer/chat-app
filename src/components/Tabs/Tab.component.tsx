import React from 'react'

interface Props {
    tabIndex: Number,
    tabName: String,
    setselected: any,
    selected: Number
}

const Tab = ({ tabIndex, tabName, setselected, selected }: Props) => {
    return (
        <div className="tab">
            <p onClick={() => setselected(tabIndex)}>{tabName}</p>
        </div>
    )
}

export default Tab
