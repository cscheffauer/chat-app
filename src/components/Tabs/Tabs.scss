@import "../../variables";

.tabs {
    width: 100%;
    height: 8vh; /* Fallback for browsers that do not support Custom Properties */
    height: calc(var(--vh, 1vh) * 8);
    display:flex;
    flex-direction: row;
    .tab{
        cursor: pointer;
        flex: 1;
        display: flex;
        align-items: center;
        justify-content: center;
        border-bottom: $border-style-tabs;
    }
    .selected {
        background-color: $color-chat-window;
        border-top-left-radius: 5px;
        border-top-right-radius: 5px;
        border-top: $border-style-tabs;
        border-left: $border-style-tabs;
        border-right: $border-style-tabs;
        border-bottom: none;
    }
}