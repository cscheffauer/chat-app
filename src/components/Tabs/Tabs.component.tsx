import React from 'react';
import './Tabs.scss';

import Tab from './Tab.component';

interface Props {
	setselected: any;
	selected: Number;
	participantNumber: Number;
}

const Tabs = ({ participantNumber, ...additionalProps }: Props) => {
	return (
		<div className='tabs'>
			<Tab tabIndex={1} tabName={'Participants (' + participantNumber + ')'} {...additionalProps} />
			<Tab tabIndex={2} tabName={'Chat'} {...additionalProps} />
		</div>
	);
};

export default Tabs;
