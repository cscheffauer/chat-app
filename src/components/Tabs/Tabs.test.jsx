import React from 'react';
import { mount } from 'enzyme';
import Tabs from './Tabs.component';

describe('<Tabs />', () => {
    let wrapper;
    const setselectedMock = jest.fn();

    beforeEach(() => {
        wrapper = mount(<Tabs setselected={setselectedMock} selected={2} participantNumber={8} />)
    });

    afterEach(() => {
        jest.clearAllMocks();
    });
    it('expect to render Tabs component', () => {
        expect(wrapper).toMatchSnapshot();
    })

    describe('Testing the Tabs', () => {
        it('expect to have the 2nd tab selected initially', () => {
            const secondTab = wrapper.find({ tabIndex: 2 })
            expect(secondTab.children().at(0).hasClass('selected')).toBe(true);
        })

        it('expect to set the selected to 1', () => {
            const firstTab = wrapper.find({ tabIndex: 1 })
            firstTab.children().at(0).props().onClick({ preventDefault() { } });

            expect(setselectedMock).toHaveBeenCalledWith(1);            //tab index 1 should be passed to the setselectedMock fn

        })

    });

})


