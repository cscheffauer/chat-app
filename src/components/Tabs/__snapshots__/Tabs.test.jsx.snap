// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs /> expect to render Tabs component 1`] = `
<Tabs
  participantNumber={8}
  selected={2}
  setselected={[MockFunction]}
>
  <div
    className="tabs"
  >
    <Tab
      selected={2}
      setselected={[MockFunction]}
      tabIndex="Userlist"
      tabName="Participants (8)"
    >
      <div
        className="tab "
        onClick={[Function]}
      >
        <p>
          Participants (8)
        </p>
      </div>
    </Tab>
    <Tab
      selected={2}
      setselected={[MockFunction]}
      tabIndex="Chat"
      tabName="Chat"
    >
      <div
        className="tab "
        onClick={[Function]}
      >
        <p>
          Chat
        </p>
      </div>
    </Tab>
  </div>
</Tabs>
`;
